testing/file_test/README.md

```diff
@@ -65,6 +64,17 @@ CARBON_FILE_TEST_FACTORY(MyFileTest);
 }  // namespace Carbon::Testing
 ```
 
+## Filename `fail_` prefixes
+
+When a run fails, information about what pieces failed are returned on
+`RunResult`. This affects whether a `fail_` prefix on the file is required,
+including in combination with split-file tests (using the `// --- <filename>`
+comment marker).
+
+The main test file and any split-files must have a `fail_` prefix if and only if
+they have an associated error. An exception is that the main test file may omit
+`fail_` when it contains split-files that have a `fail_` prefix.
```

Comment:
> Are you saying:
> 
> * If you don't use split files, then `fail_` prefix on main test file if and only if it has an error.
> * If you do use split files, then `fail_` prefix on each split that has a failure. If there is any failure then the main test file may optionally also start with `fail_`.
>   ?

Yes, that's a different way to phrase it. I prefer this phrasing because it makes behavior sound less divergent.

Reply:
Actually, there's a nuance your phrasing is missing. There can be no per-file failures, and the main test file may still fail. Then `fail_` is required.